oap-server/server-library/library-module/src/main/java/org/apache/skywalking/oap/server/library/module/ModuleProvider.java

```diff
@@ -52,9 +52,28 @@ protected final ModuleManager getManager() {
     public abstract Class<? extends ModuleDefine> module();
 
     /**
-     *
+     * Create a config creator to initialize this configuration of this module provider
+     * @return creator instance to initialize the configuration with callback. Or return null if no config is required.
      */
-    public abstract ModuleConfig createConfigBeanIfAbsent();
+    public abstract ConfigCreator newConfigCreator();
```

Comment:
There are tons of warnings (all implementations) because of the raw types, maybe add the generic type to eliminate them?

```suggestion
    public abstract ConfigCreator<? extends ModuleConfig> newConfigCreator();
```